akka-cluster/src/main/scala/akka/cluster/ClusterRemoteWatcher.scala

```diff
@@ -160,7 +160,7 @@ private[cluster] class ClusterRemoteWatcher(
     if (!clusterNodes(watchee.path.address)) super.watchNode(watchee)
 
   override protected def shouldWatch(watchee: InternalActorRef): Boolean =
-    clusterNodes(watchee.path.address)
+    clusterNodes(watchee.path.address) || super.shouldWatch(watchee)
```

Comment:
I'm pretty sure it must be changed in `RemoteWatcher` (super) to:
```scala
  @InternalApi protected def shouldWatch(@unused watchee: InternalActorRef): Boolean = {
    // In this it is unnecessary if only created by RARP, but cluster needs it.
    // Cleaner than overriding Cluster watcher addWatch/removeWatch just for one boolean test
    remoteProvider.remoteSettings.UseUnsafeRemoteFeaturesWithoutCluster
  }
```

Because  current implementation will always return `true` when when using cluster and that would be wrong if `use-unsafe-remote-features-without-cluster=off`.

Current implementation is using:
```
private[akka] final val hasClusterOrUseUnsafe = settings.HasCluster || remoteSettings.UseUnsafeRemoteFeaturesWithoutCluster
```
and `settings.HasCluster` will always be `true` for that case

Reply:
My understanding was that if using cluster, the use unsafe would not matter, because it would always be safe. Updated to use the setting.